lockbox-ios/Store/AccountStore.swift

```diff
@@ -80,14 +87,28 @@ class AccountStore {
                            redirectUri: Constant.fxa.redirectURI)
 
                     self.generateLoginURL()
-                    self.populateAccountInformation()
                 }
+
+                self._oauthInfo.onNext(nil)
+                self._profile.onNext(nil)
             }
         }
+
+        self.checkOldAccount()
     }
 }
 
 extension AccountStore {
+    private func checkOldAccount() {
+        if self.keychainWrapper.string(forKey: KeychainKey.email.rawValue) != nil ||
+                self.keychainWrapper.string(forKey: KeychainKey.displayName.rawValue) != nil ||
+                self.keychainWrapper.string(forKey: KeychainKey.avatarURL.rawValue) != nil {
+            self._oldAccountPresence.onNext(true)
+        } else {
+            self._oldAccountPresence.onNext(false)
+        }
+    }
```

Comment:
This feels like it should go in a more central place, and the upgrade should go in a `LifecycleAction.upgrade(old: Int, new: Int)` action.